components/renku_data_services/storage_schemas/api.spec.yaml

```diff
@@ -47,9 +148,112 @@ components:
       type: object
       properties:
         git_url:
+          $ref: "#/components/schemas/GitUrl"
+        storage_type:
+          type: string
+          description: same as rclone prefix
+        configuration:
+          type: object
+          description: Dictionary of rclone key:value pairs (based on schema from '/storage_schema')
+          additionalProperties:
+            type: string
+      required:
+        - git_url
+        - storage_type
+        - configuration
+    CloudStorageWithId:
+      allOf:
+        - $ref: "#/components/schemas/CloudStorage"
+        - type: object
+          required:
+            - storage_id
+          properties:
+            storage_id:
+              $ref: "#/components/schemas/UlidId"
+    RCloneSchema:
+      type: array
+      items:
+        $ref: "#/components/schemas/RCloneEntry"
+    RCloneEntry:
+      type: object
+      properties:
+        name:
+          type: string
+          description: name of the provider
+        description:
+          type: string
+          description: description of the provider
+        prefix:
+          type: string
+          description: slug name of the provider
+        options:
+          type: array
+          items:
+            $ref: "#/components/schemas/RCloneOption"
+    RCloneOption:
+      type: object
+      properties:
+        name:
+          type: string
+          description: name of the option
+        help:
+          type: string
+          description: help text for the option
+        provider:
+          type: string
+          description: The cloud provider the option is for (See 'provider' option)
+        default:
+          oneOf:
+            - type: number
+            - type: string
+            - type: boolean
+            - type: object
+          description: default value for the option
+        default_str:
           type: string
-        type:
+          description: string representation of the default value
+        examples:
+          type: array
+          items:
+            type: object
+            properties:
+              value:
+                type: string
+                description: a potential value for the option (think enum)
+              help:
+                type: string
+                description: help text for the value
```

Comment:
In the end, all of this is just a schema for the rclone generated schema because I thought it's nicer than to just have `dict[str, any]` at the top level. Even though it depends on internals of rclone. but if rclone changes those we need to adapt anyways I think.